src/ProjectTemplates/test/Helpers/ProcessEx.cs

```diff
@@ -170,20 +171,16 @@ internal string GetFormattedOutput()
             return $"Process exited with code {_process.ExitCode}\nStdErr: {Error}\nStdOut: {Output}";
         }
 
-        public void WaitForExit(bool assertSuccess, TimeSpan? timeSpan = null)
+        public async ValueTask WaitForExit(bool assertSuccess, TimeSpan? timeSpan = null)
         {
-            if(!timeSpan.HasValue)
+            if (!timeSpan.HasValue)
             {
                 timeSpan = TimeSpan.FromSeconds(600);
             }
 
-            var exited = Exited.Wait(timeSpan.Value);
-            if (!exited)
-            {
-                _output.WriteLine($"The process didn't exit within the allotted time ({timeSpan.Value.TotalSeconds} seconds).");
-                _process.Dispose();
-            }
-            else if (assertSuccess && _process.ExitCode != 0)
+            await Exited.TimeoutAfter(timeSpan.Value);
```

Comment:
Does it blow up or gracefully exit if there is a timeout? If it exits silently, will the TimeOut be logged in the StdOut?

Reply:
It throws: https://github.com/aspnet/Extensions/blob/24c84a5b1db0fca3cfcc54eed3f4f94c6670a709/src/TestingUtils/Microsoft.AspNetCore.Testing/src/TaskExtensions.cs#L33. It's a slightly different error message, but still useful.

Reply:
Interesting. This is different from what was before. Are you sure about letting it blow up for every timeout? Might be worth just logging it and be hopeful that the test still passes. There are a bunch of intermittent flakiness here that could cause random timeouts that may not affect the test.